python/requirements_compiled.txt

```diff
@@ -1519,7 +1519,7 @@ py-cpuinfo==9.0.0
     # via deepspeed
 py-partiql-parser==0.5.0
     # via moto
-py-spy==0.3.14
+py-spy==0.4.0
```

Comment:
this constraint file is used in py39 test environments, and I think this can lead to failures. I think we need to have separate requirement constraint files for different python versions in the long run. in the short run, maybe `py-spy` can be treated as a special case.